cmd/juju/controller/listmodels.go

```diff
@@ -236,119 +169,213 @@ func (c *modelsCommand) getNewModelInfo(ctx *cmd.Context, client ModelManagerAPI
+	// since these will be the same for all models here...
+	ControllerUUID     string              `json:"controller-uuid" yaml:"controller-uuid"`
+	ControllerName     string              `json:"controller-name" yaml:"controller-name"`
+	Owner              string              `json:"owner" yaml:"owner"`
+	Cloud              string              `json:"cloud" yaml:"cloud"`
+	CloudRegion        string              `json:"region,omitempty" yaml:"region,omitempty"`
+	ProviderType       string              `json:"type,omitempty" yaml:"type,omitempty"`
+	Life               string              `json:"life" yaml:"life"`
+	Status             *common.ModelStatus `json:"status,omitempty" yaml:"status,omitempty"`
+	UserAccess         string              `yaml:"access" json:"access"`
+	UserLastConnection string              `yaml:"last-connection" json:"last-connection"`
+
+	// Counts is the map of different counts where key is the entity that was counted
+	// and value is the number, for e.g. {"machines":10,"cores":3}.
+	Counts       map[string]int64 `json:"-" yaml:"-"`
+	SLA          string           `json:"sla,omitempty" yaml:"sla,omitempty"`
+	SLAOwner     string           `json:"sla-owner,omitempty" yaml:"sla-owner,omitempty"`
+	AgentVersion string           `json:"agent-version,omitempty" yaml:"agent-version,omitempty"`
+}
+
+func (c *modelsCommand) modelSummaryFromParams(info params.ModelSummary, now time.Time) (ModelSummary, error) {
+	ownerTag, err := names.ParseUserTag(info.OwnerTag)
+	if err != nil {
+		return ModelSummary{}, errors.Trace(err)
 	}
-	results, err := client.ModelInfo(tags)
+	cloudTag, err := names.ParseCloudTag(info.CloudTag)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return ModelSummary{}, errors.Trace(err)
 	}
-
-	info := []common.ModelInfo{}
-	for i, result := range results {
-		if result.Error != nil {
-			if params.IsCodeUnauthorized(result.Error) {
-				// If we get this, then the model was removed
-				// between the initial listing and the call
-				// to query its details.
-				continue
-			}
-			return nil, errors.Annotatef(
-				result.Error, "getting model %s (%q) info",
-				userModels[i].UUID, userModels[i].Name,
-			)
+	summary := ModelSummary{
+		ShortName:      info.Name,
+		Name:           jujuclient.JoinOwnerModelName(ownerTag, info.Name),
+		UUID:           info.UUID,
+		ControllerUUID: info.ControllerUUID,
+		Owner:          ownerTag.Id(),
+		Life:           string(info.Life),
+		Cloud:          cloudTag.Id(),
+		CloudRegion:    info.CloudRegion,
+		UserAccess:     string(info.UserAccess),
+	}
+	if info.AgentVersion != nil {
+		summary.AgentVersion = info.AgentVersion.String()
+	}
+	// Although this may be more performance intensive, we have to use reflection
+	// since structs containing map[string]interface {} cannot be compared, i.e
+	// cannot use simple '==' here.
+	if !reflect.DeepEqual(info.Status, params.EntityStatus{}) {
```

Comment:
Should we be making info.Status a *params.EntityStatus so we can just compare to nil instead of needing reflect?